### PR TITLE
Updated Player Event Contexts

### PR DESCRIPTION
**Pull request by Xenmai**
PlayerJumpScriptEvent was already updated. Only ChatScriptEvent still
needs to be updated, because it has a foreach.

### CODE REVIEW

File: src/main/java/net/aufdemrand/denizen/events/player/ListPingScriptEvent.java [hunk 1]
```diff
@@ -91,13 +91,20 @@ public boolean applyDetermination(ScriptContainer container, String determinatio
...
+        if (name.equals("motd")) {
+            return motd;
+        }
+        else if (name.equals("max_players")) {
+            return max_players;
+        }
+        else if (name.equals("num_players")) {
+            return num_players;
+        }
+        else if (name.equals("adress")) {
```
Comment (Fortifier42) [inline src/main/java/net/aufdemrand/denizen/events/player/ListPingScriptEvent.java]: Spelt Address wrong.

File: src/main/java/net/aufdemrand/denizen/events/player/PlayerChangesSignScriptEvent.java [hunk 2]
```diff
@@ -113,14 +113,23 @@ public ScriptEntryData getScriptEntryData() {
...
+        else if (name.equals("material")) {
+            return material;
+        }
+        else if (name.equals("new")) {
+            return new_sign;
+        }
+        else if (name.equals("old")) {
+            return old_sign;
+        }
+        else if (name.equals("cuboids")) {
```
Comment (Fortifier42) [inline src/main/java/net/aufdemrand/denizen/events/player/PlayerChangesSignScriptEvent.java]: add "// Deprecated in favor of context.location.cuboids" to the end

File: src/main/java/net/aufdemrand/denizen/events/player/PlayerDamagesBlockScriptEvent.java [hunk 3]
```diff
@@ -103,12 +103,17 @@ public ScriptEntryData getScriptEntryData() {
...
-        context.put("cuboids", cuboids);
-        return context;
+    public dObject getContext(String name) {
+        if (name.equals("location")) {
+            return location;
+        }
+        else if (name.equals("material")) {
+            return material;
+        }
+        else if (name.equals("cuboids")) {
```
Comment (Fortifier42) [inline src/main/java/net/aufdemrand/denizen/events/player/PlayerDamagesBlockScriptEvent.java]: add "// Deprecated in favor of context.location.cuboids" to the end

`else if (name.equals("cuboids")) { // Deprecated in favor of context.location.cuboids`

File: src/main/java/net/aufdemrand/denizen/events/player/PlayerFishesScriptEvent.java [hunk 4]
```diff
@@ -107,17 +107,20 @@ public ScriptEntryData getScriptEntryData() {
...
+        if (name.equals("hook")) {
+            return hook;
         }
-        if (item != null) {
-            context.put("item", item);
+        else if (name.equals("state")) {
+            return state;
         }
-        return context;
+        else if (name.equals("entity") && entity != null) {
```
Comment (Fortifier42) [inline src/main/java/net/aufdemrand/denizen/events/player/PlayerFishesScriptEvent.java]: `else if ((name.equals("entity")) && (entity != null)) {`

File: src/main/java/net/aufdemrand/denizen/events/player/PlayerFishesScriptEvent.java [hunk 5]
```diff
@@ -107,17 +107,20 @@ public ScriptEntryData getScriptEntryData() {
...
+            return entity.getDenizenObject();
+        }
+        else if (name.equals("item") && item != null) {
```
Comment (Fortifier42) [inline src/main/java/net/aufdemrand/denizen/events/player/PlayerFishesScriptEvent.java]: as above

File: src/main/java/net/aufdemrand/denizen/events/player/PlayerPlacesBlockScriptEvent.java [hunk 6]
```diff
@@ -98,13 +98,20 @@ public ScriptEntryData getScriptEntryData() {
...
-        context.put("item_in_hand", item_in_hand);
-        return context;
+    public dObject getContext(String name) {
+        if (name.equals("location")) {
+            return location;
+        }
+        else if (name.equals("material")) {
+            return material;
+        }
+        else if (name.equals("cuboids")) {
```
Comment (Fortifier42) [inline src/main/java/net/aufdemrand/denizen/events/player/PlayerPlacesBlockScriptEvent.java]: add "// Deprecated in favor of context.location.cuboids" to the end

`else if (name.equals("cuboids")) { // Deprecated in favor of context.location.cuboids`

File: src/main/java/net/aufdemrand/denizen/events/player/PlayerPlacesHangingScriptEvent.java [hunk 7]
```diff
@@ -94,12 +94,17 @@ public ScriptEntryData getScriptEntryData() {
...
-        HashMap<String, dObject> context = super.getContext();
-        context.put("hanging", hanging);
-        context.put("cuboids", cuboids);
-        context.put("location", location);
-        return context;
+    public dObject getContext(String name) {
+        if (name.equals("hanging")) {
+            return hanging;
+        }
+        else if (name.equals("cuboids")) {
```
Comment (Fortifier42) [inline src/main/java/net/aufdemrand/denizen/events/player/PlayerPlacesHangingScriptEvent.java]: add "// Deprecated in favor of context.location.cuboids" to the end

`else if (name.equals("cuboids")) { // Deprecated in favor of context.location.cuboids`

File: src/main/java/net/aufdemrand/denizen/events/player/PlayerShearsScriptEvent.java [hunk 8]
```diff
@@ -90,11 +90,11 @@ public ScriptEntryData getScriptEntryData() {
...
-    public HashMap<String, dObject> getContext() {
-        HashMap<String, dObject> context = super.getContext();
-        context.put("entity", entity);
-        context.put("state", entity); //DEPRECATED because...dumb
-        return context;
+    public dObject getContext(String name) {
+        if (name.equals("entity")) {
+            return entity;
+        }
+        return super.getContext(name);
```
Comment (Fortifier42) [inline src/main/java/net/aufdemrand/denizen/events/player/PlayerShearsScriptEvent.java]: You still need to include 'state' in here. Just cause it is Deprecated doesn't mean players might not use it.

File: src/main/java/net/aufdemrand/denizen/events/player/PlayerStepsOnScriptEvent.java [hunk 9]
```diff
@@ -99,13 +99,20 @@ public ScriptEntryData getScriptEntryData() {
...
+        if (name.equals("location")) {
+            return location;
+        }
+        else if (name.equals("previous_location")) {
+            return previous_location;
+        }
+        else if (name.equals("new_location")) {
+            return new_location;
+        }
+        else if (name.equals("cuboids")) {
```
Comment (Fortifier42) [inline src/main/java/net/aufdemrand/denizen/events/player/PlayerStepsOnScriptEvent.java]: add "// Deprecated in favor of context.location.cuboids" to the end

`else if (name.equals("cuboids")) { // Deprecated in favor of context.location.cuboids`

File: src/main/java/net/aufdemrand/denizen/events/player/PlayerFishesScriptEvent.java (hunk 4 above)
Comment (Morphan1) [inline src/main/java/net/aufdemrand/denizen/events/player/PlayerFishesScriptEvent.java]: What he had is more reasonable than that

**Comment by Morphan1** [comment]
Don't listen to @Fortifier42, your if formatting was fine.
`((blah != null) && (whateverelse))` is just unnecessary. Might as well be `(((blah != null) && (whateverelse) == true))`

**Comment by mcmonkey4eva** [comment]
Morph is correct, that is pointless, @Fortifier42 ... eh well, despite the silliness, pulled. We can clean code later. Preferably in singular massive commits :D my favorite

**Pull request merged**